client/pages/facts-edit.jsx

```diff
@@ -55,19 +35,52 @@ export default React.createClass({
 var ProjectField = React.createClass({
   propTypes: {
     project: React.PropTypes.object,
-    field: React.PropTypes.string,
-    update: React.PropTypes.func,
-    save: React.PropTypes.func
+    field: React.PropTypes.string
   },
   getInitialState () {
     let project = this.props.project
     let field = this.props.field
     let type = (project.schema[field] && project.schema[field].type) || 'text'
     let content = project.facts.json[field]
-    return { type, content }
+    let newContent = (content instanceof Object) ? Object.assign({}, content) : content
+    return { type, content, newContent }
+  },
+  isInvalid () {
+    let invalid = validator.validateDocField({
+      doc: this.props.project,
+      field: this.props.field,
+      newValue: this.state.newContent
+    })
+    if (invalid.error) {
+      this.setState({ validationError: invalid.error })
+    } else {
+      this.setState({ validationError: null })
+    }
+    return !!invalid
+  },
```

Comment:
Again, could we change this function name to `isValid`?

I think `return !!invalid` will also now not work properly since invalid will now always be an object, not to mention it effectively reads "not not not valid".